iroha/src/tx.rs

```diff
@@ -3,155 +3,76 @@ use iroha_derive::Io;
 use parity_scale_codec::{Decode, Encode};
 use std::time::SystemTime;
 
+/// This structure represents transaction in non-trusted form.
+///
+/// `Iroha` and its' clients use this structure to send via network.
+/// Direct usage in business logic is strongly prohibited.
+/// Wrap this structure in `Transaction::Requested` and
+/// go through `Transaction` lifecycle.
+#[derive(Clone, Debug, Io, Encode, Decode)]
+pub struct TransactionRequest {
+    /// Account ID of transaction creator (username@domain).
+    account_id: Id,
+    /// An ordered set of instructions.
+    pub instructions: Vec<Contract>,
```

Comment:
Maybe introduce a getter for instructions (e.g. `pub fn instructions(&self) -> &[Contract]`)? I think we don't need a write access to them.

Reply:
Idk, we have a plan to add event triggers which will produce new instructions, so let stay with it for now. Ideally we may make it private without getter and just ask methods on it